### PR TITLE
Hello World Sample for R

### PR DESCRIPTION
**Pull request by averikitsch**
## Proposed Changes

- Add community sample for R

### CODE REVIEW

File: community/samples/serving/helloworld-r/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,21 @@
+# Use the official Python image.
+# https://hub.docker.com/_/python
+FROM python:3.7
```
Comment (grayside) [inline community/samples/serving/helloworld-r/Dockerfile]: 1. Why not use something like https://shiny.rstudio.com/? Fair enough if this is a more typical pattern.

2. Kind of need to see Python and R together here, but reasons why helloworld-shell uses go:

  * Small resulting image, as the go runtime does not need to be included once the binary is produced.
  * Higher performance.

I think it's a fair counter-argument that R developers will be more likely familiar with Python.

File: community/samples/serving/helloworld-r/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+ENV APP_HOME /app
+WORKDIR $APP_HOME
+COPY . .
+
+# Install production dependencies.
+RUN pip install Flask gunicorn
+
+# Install R
+RUN apt-get update && \
+    apt-get install -y --no-install-recommends --allow-unauthenticated r-base
```
Comment (grayside) [inline community/samples/serving/helloworld-r/Dockerfile]: Why do you need `--allow-unauthenticated`?

File: community/samples/serving/helloworld-r/README.md [hunk 3]
```diff
@@ -0,0 +1,195 @@
+---
+title: "Hello World - Shell"
```
Comment (grayside) [inline community/samples/serving/helloworld-r/README.md]:
```suggestion
title: "Hello World - R"
```

File: community/samples/serving/helloworld-r/README.md [hunk 4]
```diff
@@ -0,0 +1,195 @@
...
+linkTitle: "Shell"
```
Comment (grayside) [inline community/samples/serving/helloworld-r/README.md]:
```suggestion
linkTitle: "R"
```

File: community/samples/serving/helloworld-r/README.md [hunk 5]
```diff
@@ -0,0 +1,195 @@
...
+weight: 1
+type: "docs"
+---
+
+A simple web app that executes a shell script. The shell script reads an env
```
Comment (grayside) [inline community/samples/serving/helloworld-r/README.md]: And other s/shell/R instances.

File: community/samples/serving/helloworld-r/README.md [hunk 6]
```diff
@@ -0,0 +1,195 @@
...
+
+   ```sh
+   #!/bin/sh
+   echo Hello ${TARGET:=World}!
+   ```
+
+1. Create a new file named `invoke.go` and paste the following code. We use a
+   basic web server written in Go to execute the shell script:
+
+   ```go
```
Comment (grayside) [inline community/samples/serving/helloworld-r/README.md]: Using python instead of Go

File: community/samples/serving/helloworld-r/app.py [hunk 7]
```diff
@@ -0,0 +1,15 @@
+import os
+import subprocess
+
+from flask import Flask
+
+app = Flask(__name__)
+
+@app.route('/')
+def hello_world():
+    output = subprocess.check_output('/usr/bin/Rscript HelloWorld.R', shell=True)
```
Comment (grayside) [inline community/samples/serving/helloworld-r/app.py]: If the Rscript errors out, will this respond with an HTTP 500?

File: community/samples/serving/helloworld-r/Dockerfile (hunk 1 above)
Comment (averikitsch) [reply community/samples/serving/helloworld-r/Dockerfile]: 1. Shiny/rstudio from my research, runs the studio on a server - no actual script running or server request/response capabilities. 
2. I went back and forth on this. R users are going to have no experience with Go, but they will know what Python is and looks like. Though we can make the argument that they may never have to touch the Go code for this to work.

File: community/samples/serving/helloworld-r/README.md (hunk 6 above)
Comment (averikitsch) [reply community/samples/serving/helloworld-r/README.md]: Saving files is hard sometimes.

File: community/samples/serving/helloworld-r/Dockerfile (hunk 2 above)
Comment (averikitsch) [reply community/samples/serving/helloworld-r/Dockerfile]: not necessary. removed.

File: community/samples/serving/helloworld-r/app.py (hunk 7 above)
Comment (averikitsch) [reply community/samples/serving/helloworld-r/app.py]: yes, but I updated to make it look nicer.

File: community/samples/serving/helloworld-r/README.md [hunk 8]
```diff
@@ -0,0 +1,194 @@
...
+A simple web app that executes an R script. The R script reads an env
+variable `TARGET` and prints `Hello ${TARGET}!`. If the `TARGET` environment
+variable is not specified, the script uses `World`.
+
+Follow the steps below to create the sample code and then deploy the app to your
+cluster. You can also download a working copy of the sample, by running the
+following commands:
+
+```shell
+git clone -b "release-0.7" https://github.com/knative/docs knative-docs
```
Comment (RichieEscarez) [inline community/samples/serving/helloworld-r/README.md]:
```suggestion
git clone -b "{{< branch >}}" https://github.com/knative/docs knative-docs
```
FYI: https://github.com/knative/docs/pull/1667

Comment (averikitsch) [reply community/samples/serving/helloworld-r/README.md]: done.

File: community/samples/serving/helloworld-r/README.md [hunk 9]
```diff
@@ -0,0 +1,194 @@
...
+   ENV APP_HOME /app
+   WORKDIR $APP_HOME
+   COPY . .
+
+   # Install production dependencies.
+   RUN pip install Flask gunicorn
+
+   # Install R
+   RUN apt-get update && \
+       apt-get install -y --no-install-recommends --allow-unauthenticated r-base
```
Comment (grayside) [inline community/samples/serving/helloworld-r/README.md]: Out of sync with Dockerfile.

Comment (averikitsch) [reply community/samples/serving/helloworld-r/README.md]: Thanks for catching.

**Pull request merged**